lib/travis/build/addons/jwt.rb

```diff
@@ -0,0 +1,43 @@
+require 'travis/build/addons/base'
+require 'jwt'
+
+module Travis
+  module Build
+    class Addons
+      class Jwt < Base
+        SUPER_USER_SAFE = true
+
+        def before_before_script
+          secrets = nil
+          if config.is_a?(String)
+            secrets = [config]
+          elsif config.is_a?(Array)
+            secrets = config
+          else
+            secrets = config.values
+          end
+
+          tokens = {}
+          secrets.each do |secret|
+            key, secret = secret.split('=').map(&:strip)
+            pull_request = self.data.pull_request ? self.data.pull_request : ""
+            now = Time.now.to_i()
+            payload = {"iss" => "travis-ci.org",
```

Comment:
Or even a way to pull more useful ones? Like what product/domain we are on?

On Jul 26, 2016 12:10 PM, "Hiro Asari" <notifications@github.com> wrote:

> In lib/travis/build/addons/jwt.rb
> <https://github.com/travis-ci/travis-build/pull/356#discussion_r72316706>:
>
> > +        def before_before_script
> > +          secrets = nil
> > +          if config.is_a?(String)
> > +            secrets = [config]
> > +          elsif config.is_a?(Array)
> > +            secrets = config
> > +          else
> > +            secrets = config.values
> > +          end
> > +
> > +          tokens = {}
> > +          secrets.each do |secret|
> > +            key, secret = secret.split('=').map(&:strip)
> > +            pull_request = self.data.pull_request ? self.data.pull_request : ""
> > +            now = Time.now.to_i()
> > +            payload = {"iss" => "travis-ci.org",
>
> Good question. I don't know if that is static, or what.
>
> @joshk <https://github.com/joshk> Do you have opinions?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/travis-ci/travis-build/pull/356/files/b5a1fb6de21ee237ca552a761d612899bf11ee0f#r72316706>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGuByOI1bewjjHY_zobh5HBu5fpKO8Pks5qZluagaJpZM4DHeYE>
> .
>